frontend/src/modules/Survey/Components/Survey.tsx

```diff
@@ -6,51 +6,69 @@ import { StepBegin } from 'Survey/Components/StepBegin'
 import { StepCourse } from 'Survey/Components/StepCourse'
 import { StepRadio } from 'Survey/Components/StepRadio'
 import { StepFinal } from 'Survey/Components/StepFinal'
-import {
-  sendSurveyData,
-  SurveyData,
-} from 'Survey/Components/FuncsAndConsts/SurveyFunctions'
+import { SurveyData } from 'Survey/Components/FuncsAndConsts/SurveyFunctions'
 import { Question } from '@core/Types'
+import axios from 'axios'
+import { API_ROOT, STUDENT_API } from '@core/Constants'
 
 export const Survey = () => {
   const [showError, setShowError] = useState(false)
-  const [currStep, setCurrStep] = useState(0)
+  const [currStep, setCurrStep] = useState(1)
+  const [surveySuccess, setSurveySuccess] = useState(false)
+  const [surveyError, setSurveyError] = useState('')
+
   // If there are custom questions the below will be a network call perhaps
   const questions: Question[] = require('@core/Questions/Questions.json')
   const numSpecialQuestions = 1 // Course list
-  const totalSteps = questions.length + numSpecialQuestions
+  const totalSteps = questions.length + numSpecialQuestions + 1
 
   // Form answer props
   const [nameAnswer, setNameAnswer] = useState('')
   const [emailAnswer, setEmailAnswer] = useState('')
   const [courseList, setCourseList] = useState<string[]>([])
-  const [answers, setAnswers] = useState(Array<string>(questions.length).fill('')) // Will be in order of Qs
+  const [answers, setAnswers] = useState(
+    Array<string>(questions.length).fill('')
+  ) // Will be in order of Qs
 
   const changeAnswer = (i: number, v: string) => {
-    setAnswers(answers.map((value, index) => index === i ? v : value))
+    setAnswers(answers.map((value, index) => (index === i ? v : value)))
   }
 
   // last step's Next button handles sending data
   function finalNext() {
-    const mcData = Object.fromEntries(questions.map((question, index) => [question.questionId, answers[index]]))
+    const mcData = Object.fromEntries(
+      questions.map((question, index) => [question.questionId, answers[index]])
+    )
     const surveyData: SurveyData = {
       courseCatalogNames: courseList,
       name: nameAnswer,
       email: emailAnswer,
-      ...mcData
+      ...mcData,
     }
     console.log('Finished survey', surveyData)
-    sendSurveyData(surveyData)
-    setCurrStep(currStep + 1)
+    axios.post(`${API_ROOT}${STUDENT_API}/survey`, surveyData).then(
+      (response: any) => {
+        console.log(response)
+        setSurveySuccess(true)
+        setCurrStep(currStep + 1)
+      },
+      (error: any) => {
+        console.log(error)
+        setSurveySuccess(false)
+        setSurveyError(error.message)
```

Comment:
You likely want to use the error message provided in the response, not the message on the error object. For example right now when submitting an invalid course, the error displayed is

> "Error: Request failed with status code 400"

instead of

> "Could not find course data for class CS2114"

which is in the response message. Maybe https://axios-http.com/docs/handling_errors can help